src/main/java/com/exasol/releaserobot/usecases/Repository.java

```diff
@@ -0,0 +1,148 @@
+package com.exasol.releaserobot.usecases;
+
+import java.io.*;
+import java.util.*;
+
+import org.kohsuke.github.*;
+
+import com.exasol.releaserobot.repository.*;
+
+/**
+ * This class represents Git repository content based on the latest commit of the user-specified branch.
+ */
+public abstract class Repository {
+    private static final String CHANGELOG_FILE_PATH = "doc/changes/changelog.md";
+    private final GHRepository ghRepository;
```

Comment:
This should not be here. This is your internal model and should not know anything about Github.

The `RepositoryGateway` should give you a back valid `Repository` and make the right instantiation/transformation.

Reply:
It retrieves files as strings that I need for some checks. Should we store all the files in an object?

Reply:
We agreed to do it in another PR